tests/utils/test_progress_bar.py

```diff
@@ -134,6 +134,63 @@ def test_remote(i, progress_bar: RemoteProgressBar):
         dis.refresh()
         time.sleep(random.random() / 10)
 
+from ablator.utils.progress_bar import in_notebook
+def test_in_notebook():
+    result=in_notebook()
+    assert result==False,"The in_notebook function cannot correctly determine whether it is a terminal."
+    with mock.patch('ablator.utils.progress_bar.in_notebook', side_effect=ImportError):
```

Comment:
I do not think this is a good test to see if it is robust in Notebook. The import errors could be related to other issues 
```
if "IPKernelApp" not in get_ipython().config:  # pragma: no cover
    return False
```

we should find a way to simulate that the ipykernel is running. Will require significant research.